src/runtime/TemporalObject.cpp

```diff
@@ -1438,6 +1512,210 @@ std::map<TemporalObject::DateTimeUnits, int> TemporalTimeZoneObject::getISOParts
     return result;
 }
 
+bool TemporalTimeZoneObject::isValidTimeZoneName(const std::string& timeZone)
+{
+    return timeZone == "UTC";
+}
+
+std::string TemporalTimeZoneObject::canonicalizeTimeZoneName(const std::string& timeZone)
+{
+    return "UTC";
+}
+
+Value TemporalTimeZoneObject::createTemporalTimeZone(ExecutionState& state, const std::string& identifier, Optional<Object*> newTarget)
+{
+    if (TemporalTimeZoneObject::isValidTimeZoneName(identifier)) {
+        ASSERT(TemporalTimeZoneObject::canonicalizeTimeZoneName(identifier) == identifier);
+        return new TemporalTimeZoneObject(state, state.context()->globalObject()->temporal()->asTemporalObject()->getTemporalTimeZonePrototype(), identifier, Value());
+    }
+
+    int64_t offsetNanoseconds = TemporalInstantObject::offsetStringToNanoseconds(state, identifier);
+
+    return new TemporalTimeZoneObject(state, state.context()->globalObject()->temporal()->asTemporalObject()->getTemporalTimeZonePrototype(), TemporalTimeZoneObject::formatTimeZoneOffsetString(offsetNanoseconds), Value(offsetNanoseconds));
+}
+
+std::string TemporalTimeZoneObject::formatTimeZoneOffsetString(long long offsetNanoseconds)
+{
+    std::string result = offsetNanoseconds >= 0 ? "+" : "-";
+    offsetNanoseconds = std::abs(offsetNanoseconds);
+    int nanoseconds = offsetNanoseconds % TemporalInstantObject::SecondToNanosecond;
+    int seconds = int(std::floor(offsetNanoseconds / TemporalInstantObject::SecondToNanosecond)) % 60;
+    int minutes = int(std::floor(offsetNanoseconds / TemporalInstantObject::MinuteToNanosecond)) % 60;
+    int hours = int(std::floor(offsetNanoseconds / TemporalInstantObject::HourToNanosecond));
+
+    result += (hours == 0 ? "00" : ((hours < 10 ? "0" : "") + std::to_string(hours))) + ":" + (minutes < 10 ? "0" : "") + std::to_string(minutes);
+
+    if (nanoseconds != 0) {
+        std::ostringstream ss;
+        ss << std::setw(9) << std::setfill('0') << nanoseconds;
```

Comment:
How about this method
?
We can figure out the digit number of `nanoseconds` below
```c++
int num = nanoseconds;
int digitNum = 0;
while(num != 0) {
    num = num/10;
    ++digitNum;
}
```

Convert `nanoseconds` to string using `itoa` function (stdlib is already included in Escargot) and fill the rest digit as 0 using the calculated digit number.

Reply:
Thank you!